### PR TITLE
core: Add new CXI endpoint protocol FI_PROTO_CXI_RNR

### PR DESCRIPTION
**Pull request by swelch**
Reserve new Receiver-Not-Ready endpoint protocol for CXI provider.

### CODE REVIEW

File: man/fi_endpoint.3.md [hunk 1]
```diff
@@ -753,6 +753,15 @@ protocol value set to one.
...
   used by the sm2 provider
 
+*FI_PROTO_CXI*
+: Reliable-datagram protocol optimized for HPC applications
+  used by cxi provider.
+
+*FI_PROTO_CXI_RNR*
+: A version of the FI_PROTO_CXI protocol that implements an RNR
+  protocol which can be used when messaging is primarily expected
+  and FI_ORDER_SAS ordering is not required.
```
Comment (shefty) [inline man/fi_endpoint.3.md:763]: Is this a different protocol or a slight modification to the CXI protocol based on the endpoint attributes?  You call out SAS ordering, so I'm wondering if that is sufficient for the endpoint to determine if RNR messages can be sent.  What actually happens if one side chooses CXI but the other CXI_RNR?  Can they still talk?

Comment (swelch) [reply man/fi_endpoint.3.md:763]: This is a significantly different protocol with different hardware match bits formats and messaging protocols that are supported in addition to the ordering changes. While the FI_PROTO_CXI supports FI_ORDER_SAS it does not require it; it offers many other features that this new protocol does not (rendezvous messaging for instance).

If the two sides select different protocols, they cannot physically communicate. The protocols map to a different receive portals table entries, so they are physically disconnected. If a process selected this protocol and targeted an EP that did not use the protocol, the sender would get an error completion without interaction from software at the target.

**Review by shefty — APPROVED** [review]

**Comment by swelch** [comment]
@j-xiong @jswaro  I don't see how the CI failures could be related to this PR, are you OK for a merge?

**Comment by j-xiong** [comment]
The ci failure is unrelated.

**Pull request merged**